src/rgen/Microsoft.Macios.Generator/Attributes/BindFromData.cs

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.Macios.Generator.Attributes;
+
+readonly struct BindFromData : IEquatable<BindFromData> {
+
+	public string Type { get; }
+	public string? OriginalType { get; }
+
+	public BindFromData (string type)
+	{
+		Type = type;
+	}
+
+	public BindFromData (string type, string? originalType)
+	{
+		Type = type;
+		OriginalType = originalType;
+	}
+
+
+	public static bool TryParse (AttributeData attributeData,
+		[NotNullWhen (true)] out BindFromData? data)
```

Comment:
being curious. your generator targets netstandard2.1 ? I thought it was limited to 2.0, but never tried to update in a long time

Reply:
Code generators, if you do not need the integration with the IDE, can run using other versions of dotnet standard: https://github.com/dotnet/docs/commit/fa21a4bd78faf4cf28163df04d990694875be218

The tldr; they work in the compiler when using the dotnet tools, they will have problems with certain IDEs, specially VS. My approach has been, use the latests to get all the nullability checks etc.. and then make it compatible with the IDE.